Clean up some bits of UIImageView+AFNetworking.m

rdignard08 opened the pull request:
The largest change was setting the underlying dispatch queue to not be the main thread, which should help make the UI feel buttery smooth.

Additionally:
1.  `-[NSMutableDictionary setValue:forKey:]` can handle the value being `nil` unlike `-setObject...`
2.  The static const char\* keys seem needlessly verbose considering these are internal to the class(es) anyway
3.  I moved `[[UIImageView af_sharedImageCache] cacheImage:response forKey:key]` to the single standard completion block instead of that call being duplicated multiple times (and thus called multiple times)
4.  Lines 198/200/204 make use of the weakSelf/strongSelf semantic so that the UIImageView can safely dealloc when appropriate

Supernats commented on AFNetworking/UIImageView+AFNetworking.m:
```diff
@@ -148,15 +150,8 @@ - (void) removeCompletionBlocks;
-        objc_setAssociatedObject(self, &kExecutionBlocksKey, _completionBlocks, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
-    }
-    return _completionBlocks;
+    return objc_getAssociatedObject(self, _cmd) ?: objc_setAssoiciatedObject_ret(self, _cmd, [NSMutableArray new], OBJC_ASSOCIATION_RETAIN_NONATOMIC);
```
`objc_setAssoiciatedObject`?

Commit:
Clean up some bits of UIImageView+AFNetworking.m

The largest change was setting the underlying dispatch queue to not be
the main thread, which should help make the UI feel buttery smooth.

Additionally:
1. -[NSMutableDictionary setValue:forKey:] can handle the value being
nil unlike -setObject...
2. The static const char* keys seem needlessly verbose considering
these are internal to the class(es) anyway
3. I moved [[UIImageView af_sharedImageCache] cacheImage:response
forKey:key] to the single standard completion block instead of that
call being duplicated multiple times (and thus called multiple times)
4. Lines 198/200/204 make use of the weakSelf/strongSelf semantic so
that the UIImageView can safely dealloc when appropriate

tcirwin commented on AFNetworking/UIImageView+AFNetworking.m:
```diff
@@ -188,33 +187,29 @@ static void(^standardCompletionBlock)(AFHTTPRequestOperation*, id) = ^(AFHTTPReq
-                    if (success) {
+            __strong typeof(self) strongSelf = weakSelf;
+            if (op == strongSelf.af_imageRequestOperation) {
+                strongSelf.af_imageRequestOperation = nil;
```
can we wrap the whole block inside of the dispatch? (non-blocking)

rdignard08 replied:
definitely an option; I felt it was more performant to not invoke the context switch (i.e. `dispatch_async`) until we're absolutely positive we're going to use the response image.

Imagine an image view that requests an image then gets reused and requests a new image.  When the first request returns it would switch over to the main thread only to realize it's image isn't needed anymore.

tcirwin replied:
i was worried about threading issues but since we're always on the same thread here it shouldn't be a problem.

hijoumake commented on AFNetworking/UIImageView+AFNetworking.m:
```diff
@@ -68,7 +70,8 @@ + (NSOperationQueue*) af_sharedImageRequestOperationQueue {
     dispatch_once(&onceToken, ^{
         _af_imageRequestOperationQueue = [NSOperationQueue new];
-        [_af_imageRequestOperationQueue setMaxConcurrentOperationCount:NSOperationQueueDefaultMaxConcurrentOperationCount];
+        _af_imageRequestOperationQueue.underlyingQueue = dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0);
```
This is an iOS 8+ only property. There needs to be a fallback for older versions or an NSInvalidArgumentException will be thrown because the selector is not recognized.
